packages/core/contracts/proxy-scripts/lsp-broker/LspUniswapV2Broker.sol

```diff
@@ -0,0 +1,90 @@
+
+import "../../financial-templates/long-short-pair/LongShortPair.sol";
+
+/**
+ * @title LspUniswapV2Broker
+ * @notice Helper contract to facilitate batched LSP and UniswapV2 transactions, including Mint+Sell and Mint+LP.
+ */
+contract LspUniswapV2Broker {
+    using FixedPoint for FixedPoint.Unsigned;
+
+    /**
+     * @notice Mint long and short tokens and deposit them all into a UniV2 Pool.
+     */
+    function atomicMintAddLiquidity() public {
+        /* TODO */
+    }
+
+    /**
+     * @notice Mint long and short tokens and convert all of one side into the other.
+     * @param tradingAsEOA If True, caller has balance of collateral and expects to receive back all long/short tokens.
+     * @param sellLong If True, converts all long tokens into short, else the opposite.
+     * @param longShortPair LSP contract address to mint position on.
+     * @param router Contract to call to exchange long and short tokens.
+     * @param collateralToMintWith Amount of collateral to deposit and borrow long and short tokens against.
+     * @param swapPath `Router.swapExactTokensForTokens` param: path with which to swap token to sell for the other.
+     * @param deadline `Router.swapExactTokensForTokens` param: time before transaction must be mined.
+     */
+    function atomicMintSellOneSide(
+        bool tradingAsEOA,
+        bool sellLong,
+        LongShortPair longShortPair,
+        IUniswapV2Router01 router, /* TODO: Should we allow `router` to be any exchange, such as a Matcha multihop? */
+        uint256 collateralToMintWith,
+        address[] memory swapPath,
+        uint256 deadline
+    ) public {
+        require(address(longShortPair) != address(0), "Invalid long short pair");
+        require(address(router) != address(0), "Invalid router");
+        require(collateralToMintWith != 0, "Collateral to mint with");
+
+        // 0) Pull collateral from caller if necessary and approve LSP to spend it.
+        if (tradingAsEOA)
+            TransferHelper.safeTransferFrom(
+                address(longShortPair.collateralToken()),
+                msg.sender,
+                address(this),
+                collateralToMintWith
+            );
+
+        TransferHelper.safeApprove(
+            address(longShortPair.collateralToken()),
+            address(longShortPair),
+            collateralToMintWith
+        );
+
+        // 1) Deposit collateral into LSP and mint long and short tokens.
+        // TODO: So this assumes we are minting collateral --> long/short tokens in a one-to-one ratio?
+        // i.e. The `LSP.collateralPerPair=1`
+        longShortPair.create(collateralToMintWith);
+
+        // 2) Determine which token we are selling and convert it all into the other.
+        IERC20 soldToken = IERC20(sellLong ? longShortPair.shortToken() : longShortPair.longToken());
+        // TODO: Should we make this check?
+        // require(soldToken.balanceOf(address(this)) > 0, "Missing balance of token to sell");
```

Comment:
Will `swapExactTokensForTokens` revert if the amount to sell is 0? If not, then its probably worth reverting early

Reply:
it will I believe. however, there are not exactly any downsides if you try call this method and mint 0 with `collateralToMintWith=0`

Reply:
it looks like you actually added a test for this with `// Cannot mint with 0 collateral` which will sufficiently validate this I believe.

Reply:
Yeah I'm more so thinking about the situation where `create` somehow creates 0 tokens, but I suppose not reverting is not an issue

Reply:
This [PR](https://github.com/UMAprotocol/protocol/pull/3132) now prevents this edge case where you could create 0 tokens and try to swap 0 tokens.